### DISCUSSION TITLE
Logout Endpoint could be an "Authentication Flow"

### DISCUSSION THREAD

**Discussion by bs-matil** [keycloak/keycloak#13165] [Ideas]
Right now we have some "problems" migrating to the new logout endpoint of kc18+. 

While this I had the following question:

- Why is the logout page which is shown to the user not an authentication flow?
- Why is the logout flow not as configurable as the login flow
- Why do we use different apis here at all? 

What I would like to ask is. Wouldn't it be a lot more consistent to implement the logout behaviour analogue to the login behaviour. 
E.g. be able to configure additional authenticators for the logout flow. Also this would enable custom implementations for the logout. E.g. one could create an "legacy" logout authenticator which does just act as in the past while the new default logout authenticator uses the new oidc standard behaviour + one could add more steps to the flow (legacy backchannel logout etc). 

What do you think?

**Comment by ethlo** [comment]
We need this! There are so many issues, especially with IDP for logout that this would be an awesome addition. Thanks for raising this!
